pkg/asset/manifests/cloudproviderconfig.go

```diff
@@ -93,22 +91,17 @@ func (cpc *CloudProviderConfig) Generate(dependencies asset.Parents) error {
 			return nil
 		}
 		cm.Data[cloudProviderConfigCABundleDataKey] = trustBundle
+
 	case openstacktypes.Name:
-		cloud, err := icopenstack.GetSession(installConfig.Config.Platform.OpenStack.Cloud)
+		cloudProviderConfigData, cloudProviderConfigCABundleData, err := openstackmanifests.GenerateCloudProviderConfig(*installConfig.Config)
```

Comment:
Why not just pass the pointer to the install config rather than dereferencing it here?

Reply:
By passing a copy of the struct instead of a pointer, I make it clear that we expect `GenerateCloudProviderConfig` to make no modification to `installConfig.Config`. I see that as "documentation in code".

Another advantage of dereferencing early is that I don't carry around the itchy `nil` possibility any deeper.

Also I have a personal adversity for passing pointers around, to which I would stick unless there's more pressing business need?

Reply:
Note that this is a cosmetic change I am not wedded to

Reply:
I think you are going to be hard-pressed holding that line in go, but I have no objection to you doing it here.